staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -267,6 +273,10 @@ func (mounter *Mounter) Unmount(target string) error {
 	command := exec.Command("umount", target)
 	output, err := command.CombinedOutput()
 	if err != nil {
+		if err.Error() == errNoChildProcesses {
+			// We don't consider this an error (see - k/k issue #103753)
+			return nil
```

Comment:
How we make sure this unmount must be successful at this point?

Reply:
Good catch.

So exec:CombinedOutput() essentially runs [exec:Run()](https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/exec/exec.go;l=567;drc=refs%2Ftags%2Fgo1.16.6) which does this:

```
func (c *Cmd) Run() error {
	if err := c.Start(); err != nil {
		return err
	}
	return c.Wait()
}
```

This error can only come from `c.Wait()` which mean c.Start() went through without any errors. Within c.Wait(), we run the following code:

```
func (c *Cmd) Wait() error {
	if c.Process == nil {
		return errors.New("exec: not started")
	}
	if c.finished {
		return errors.New("exec: Wait was already called")
	}
	c.finished = true

	state, err := c.Process.Wait()
	if c.waitDone != nil {
		close(c.waitDone)
	}
	c.ProcessState = state

	var copyError error
	for range c.goroutine {
		if err := <-c.errch; err != nil && copyError == nil {
			copyError = err
		}
	}

	c.closeDescriptors(c.closeAfterWait)

	if err != nil {
		return err
	} else if !state.Success() {
		return &ExitError{ProcessState: state}
	}

	return copyError
}
```

And this `errNoChildProcesses` (i.e [ECHILD error](https://github.com/kubernetes/kubernetes/blob/master/vendor/golang.org/x/sys/unix/zerrors_linux_amd64.go#L659)) comes specifically from this line `state, err := c.Process.Wait()` which internally makes an [os syscall](https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/exec_unix.go;l=16;drc=refs%2Ftags%2Fgo1.16.6). But we seem to be returning that error before returning the error for the process exit state:

```
	if err != nil {
		return err
	} else if !state.Success() {
		return &ExitError{ProcessState: state}
	}
```

So additionally if we check that [c.ProcessState](https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/exec/exec.go;l=134;drc=refs%2Ftags%2Fgo1.16.6) is a success, then we should be good. Like this:

```
if err.Error() == errNoChildProcesses && command.ProcessState.Success() {
      // We don't consider this an error (see - k/k issue #103753)
      return nil
}
```

Does the above sgty?

Reply:
I've updated it to a slightly better version actually to help expose the actual process error when we get `errNoChildProcesses`:

```
		if err.Error() == errNoChildProcesses {
			if command.ProcessState.Success() {
				// We don't consider errNoChildProcesses an error if the process itself succeeded (see - k/k issue #103753).
				return nil
			}
			// Rewrite err with the actual exit error of the process.
			err = &exec.ExitError{ProcessState: command.ProcessState}
		}
```